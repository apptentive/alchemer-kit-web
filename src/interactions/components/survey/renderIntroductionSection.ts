/**
 * Renders the survey introduction text.
 *
 * @param {string} introText - The text content for the introduction section
 * @returns {HTMLDivElement} - The DOM Node
 */
const renderIntroductionSection = (introText: string) => {
  const intro = document.createElement('div');
  intro.className = 'apptentive-survey-intro';
  intro.textContent = introText;

  return intro;
};

export default renderIntroductionSection;
