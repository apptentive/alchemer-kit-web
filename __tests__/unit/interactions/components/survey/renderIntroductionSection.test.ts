import renderIntroductionSection from '../../../../../src/interactions/components/survey/renderIntroductionSection';

const _introText = 'Introduction';

describe('renderIntroductionSection', () => {
  test('properly renders error message container', () => {
    const output = renderIntroductionSection(_introText);

    expect(output).toMatchSnapshot();
  });
});
