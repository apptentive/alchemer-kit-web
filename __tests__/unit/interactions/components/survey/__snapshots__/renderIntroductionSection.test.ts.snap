// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderIntroductionSection properly renders error message container 1`] = `
<div
  class="apptentive-survey-intro"
>
  Introduction
</div>
`;
